google-ads/src/test/java/com/google/ads/googleads/lib/callables/ExceptionTransformingServerStreamingCallableTest.java

```diff
@@ -0,0 +1,103 @@
+package com.google.ads.googleads.lib.callables;
+
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Mockito.verify;
+
+import com.google.ads.googleads.lib.catalog.GeneratedCatalog;
+import com.google.ads.googleads.v5.errors.GoogleAdsError;
+import com.google.ads.googleads.v5.errors.GoogleAdsException;
+import com.google.ads.googleads.v5.errors.GoogleAdsFailure;
+import com.google.api.gax.grpc.GrpcCallContext;
+import com.google.api.gax.grpc.GrpcStatusCode;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.ServerStreamingCallable;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.grpc.StatusException;
+import java.util.Objects;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ExceptionTransformingServerStreamingCallableTest {
+
+  @Mock public ResponseObserver<Object> innerObserver;
+  private MockServerStreamingCallable innerCallable;
+
+  private static final GoogleAdsFailure googleAdsFailure =
+      GoogleAdsFailure.newBuilder().addErrors(GoogleAdsError.getDefaultInstance()).build();
+  private final Object request = "test";
+  private final ApiCallContext context = GrpcCallContext.createDefault();
+  private ExceptionTransformingServerStreamingCallable<Object, Object> serverStreamingCallable;
+  private ApiException rawApiException;
+
+  @Before
+  public void setup() {
+    this.innerCallable = new MockServerStreamingCallable();
+    this.serverStreamingCallable =
+        new ExceptionTransformingServerStreamingCallable(
+            innerCallable, new GoogleAdsExceptionTransformation());
+
+    Metadata trailers = new Metadata();
+    trailers.put(
+        GeneratedCatalog.getDefault().getLatestVersion().getExceptionFactory().getTrailerKey(),
+        googleAdsFailure.toByteArray());
+
+    rawApiException =
+        new ApiException(
+            new StatusException(Status.INVALID_ARGUMENT, trailers),
+            GrpcStatusCode.of(Code.INVALID_ARGUMENT),
+            false);
+  }
+
+  @Test
+  public void transformsExceptions() {
```

Comment:
We're testing the nice path here, but are there cases where the API will place an exception that is not transform-able in the trailing metadata key? If so, we should test for that here as well.

Reply:
Added some test cases.